File: 0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        vector<int> arr(26, 0);

        for (char c : magazine) {
            arr[c - 'a']++;
        }

        for (char c : ransomNote) {
            int index = c - 'a';
            if (arr[index] > 0) {
                arr[index]--;
            } else {
                return false;
            }
        }
        return true;
    }
};

File: 0383-ransom-note/0383-ransom-note.go
func canConstruct(ransomNote string, magazine string) bool {
    arr := [26]int{}
    
    for _, ch := range(magazine) {
        arr[ch - 'a']++
    }

    for _, ch := range(ransomNote) {
        index := ch - 'a'
        if arr[index] > 0 {
            arr[index]--
        } else {
            return false
        }
    }

    return true
}